File: blipapi_background.php
<?php

/**
 * Blip! (http://blip.pl) communication library.
 *
 * @author Marcin Sztolcman <marcin /at/ urzenia /dot/ net>
 * @version 0.02.20
 * @version $Id$
 * @copyright Copyright (c) 2007, Marcin Sztolcman
 * @license http://opensource.org/licenses/gpl-license.php GNU Public License v.2
 * @package blipapi
 */

/**
 * Blip! (http://blip.pl) communication library.
 *
 * @author Marcin Sztolcman <marcin /at/ urzenia /dot/ net>
 * @version 0.02.20
 * @version $Id$
 * @copyright Copyright (c) 2007, Marcin Sztolcman
 * @license http://opensource.org/licenses/gpl-license.php GNU Public License v.2
 * @package blipapi
 */

if (!class_exists ('BlipApi_Background')) {
    class BlipApi_Background extends BlipApi_Abstract implements IBlipApi_Command {
        protected $_user    = '';
        protected $_image   = '';

        protected function __set_user ($value) {
            $this->_user = $value;
        }

        protected function __set_image ($value) {
            $this->_image = $this->__validate_file ($value);
        }

        /**
        * Get info about users background
        *
        * @param string $user
        * @access public
        * @return array parameters for BlipApi::__query
        */
        public function read () {
            if (!$this->_user) {
                throw new InvalidArgumentException ('User name is missing.', -1);
            }
            return array ("/users/$this->_user/background", 'get');
        }

        /**
        * Upload new background
        *
        * Throws InvalidArgumentException if background path is missing, or file not found
        *
        * @param string $background new background path
        * @access public
        * @return array parameters for BlipApi::__query
        */
        public function update () {
            if (!$this->_image) {
                throw new InvalidArgumentException ('Background path is missing or file not found.', -1);
            }
            return array ('/background', 'post', array ('background[file]' => '@'.$this->_image), array ('multipart' => 1));
        }

        /**
        * Delete background
        *
        * @access public
        * @return array parameters for BlipApi::__query
        */
        public function delete () {
            return array ('/background', 'delete');
        }
    }
}

File: py/blipapi_background.py
# -*- coding: utf-8 -*-
# $Id$
#
# Blip! (http://blip.pl) communication library.
# Author: Marcin Sztolcman <marcin /at/ urzenia /dot/ net>
# Version: 0.02.03
# Copyright: (r) 2009 Marcin Sztolcman
# License: http://opensource.org/licenses/gpl-license.php GNU Public License v.2

import os.path

from blipapi__utils import prepare_post_field

def read (user):
    """ Get specified user's background info """
    if not user:
        raise ValueError ('User name is missing.')
    return ('/users/' + str (user) + '/background', 'get', None, None)

def update (background):
    """ Update current user background """
    if not os.path.isfile (background):
        raise ValueError ('Background path is missing or file not found.')

    opts = dict ()

    multipart, boundary = prepare_post_field ('background[file]', background, is_file=True)

    opts['multipart']   = multipart
    opts['boundary']    = boundary

    return ('/background', 'post', None, opts)

def delete ():
    """ Delete current user background """
    return ('/background', 'delete', None, None)

